src/Behat/Behat/Hook/Context/Attribute/HookAttributeReader.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Hook\Context\Attribute;
+
+use Behat\Behat\Context\Annotation\DocBlockHelper;
+use Behat\Behat\Context\Attribute\AttributeReader;
+use Behat\Hook\AfterFeature;
+use Behat\Hook\AfterScenario;
+use Behat\Hook\AfterStep;
+use Behat\Hook\BeforeFeature;
+use Behat\Hook\BeforeScenario;
+use Behat\Hook\BeforeStep;
+use Behat\Hook\Hook;
+use ReflectionMethod;
+
+class HookAttributeReader implements AttributeReader
+{
+    /**
+     * @var string[]
+     */
+    private static $classes = array(
```

Comment:
See comment above. I'd rather be consistent with the rest of the codebase than use "bleeding edge" in a single spot.

Though I agree a constant would be better, but as above, I think it would be better to have another PR that replaces it throughout the entire codebase.

Reply:
well, refactoring existing code requires some work (and for things that are not private, it also requires a major version).
Here, I would introduce it as a `private const`. There is no issue with introducing those progressively (for the array syntax, I agree that it is better to be consistent, as that is a pure coding standard thing, it does not impact the structure of the code)

Reply:
Okay, fair enough. I'll change it.

Reply:
Changed to `KNOWN_ATTRIBUTES`. Sound more descriptive than `$classes` anyways 🙂